File: task-service/src/main/java/com/todoapp/taskservice/health/HealthController.java
package com.todoapp.taskservice.health;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.sql.DataSource;
import java.sql.Connection;
import java.time.LocalDateTime;
import java.util.HashMap;
import java.util.Map;

@RestController
public class HealthController {

    @Autowired
    private DataSource dataSource;

    @GetMapping("/health")
    public ResponseEntity<Map<String, Object>> health() {
        return readiness();
    }

    @GetMapping("/health/live")
    public ResponseEntity<Map<String, Object>> liveness() {
        Map<String, Object> response = new HashMap<>();
        response.put("status", "UP");
        response.put("timestamp", LocalDateTime.now());
        response.put("checks", new HashMap<String, String>());
        
        return ResponseEntity.ok(response);
    }

    @GetMapping("/health/ready")
    public ResponseEntity<Map<String, Object>> readiness() {
        Map<String, Object> response = new HashMap<>();
        Map<String, String> checks = new HashMap<>();
        
        response.put("timestamp", LocalDateTime.now());
        
        // Check database connectivity
        try (Connection connection = dataSource.getConnection()) {
            if (connection.isValid(5)) {
                checks.put("database", "OK");
                response.put("status", "UP");
            } else {
                checks.put("database", "FAIL: Invalid connection");
                response.put("status", "DOWN");
            }
        } catch (Exception e) {
            checks.put("database", "FAIL: " + e.getMessage());
            response.put("status", "DOWN");
        }
        
        response.put("checks", checks);
        
        if ("DOWN".equals(response.get("status"))) {
            return ResponseEntity.status(503).body(response);
        }
        
        return ResponseEntity.ok(response);
    }
}

File: project-service/src/health/healthController.js
const mysql = require('mysql2/promise');
const consul = require('consul')();

class HealthController {
    constructor(dbConfig) {
        this.dbConfig = dbConfig;
    }

    async liveness(req, res) {
        const response = {
            status: 'UP',
            timestamp: new Date().toISOString(),
            checks: {}
        };
        
        res.json(response);
    }

    async readiness(req, res) {
        const response = {
            status: 'UP',
            timestamp: new Date().toISOString(),
            checks: {}
        };

        try {
            // Check database connectivity
            const connection = await mysql.createConnection(this.dbConfig);
            await connection.ping();
            await connection.end();
            
            response.checks.database = 'OK';
        } catch (error) {
            response.status = 'DOWN';
            response.checks.database = `FAIL: ${error.message}`;
            return res.status(503).json(response);
        }

        res.json(response);
    }

    async health(req, res) {
        return this.readiness(req, res);
    }
}

module.exports = HealthController;